File: android_mobile/src/main/java/com/alexstyl/specialdates/addevent/SelectedEvents.java
package com.alexstyl.specialdates.addevent;

import android.annotation.SuppressLint;

import com.alexstyl.specialdates.date.Date;
import com.alexstyl.specialdates.events.Event;
import com.alexstyl.specialdates.events.peopleevents.EventType;

import java.util.Collection;
import java.util.HashMap;

public final class SelectedEvents {

    private final HashMap<Integer, Event> mappedEvents;

    @SuppressLint("UseSparseArrays")
    SelectedEvents() {
        mappedEvents = new HashMap<>();
    }

    public Collection<Event> getEvents() {
        return mappedEvents.values();
    }

    void replaceDate(EventType eventType, Date date) {
        int eventId = eventType.getId();
        Event event = new Event(eventType, date);
        mappedEvents.put(eventId, event);
    }

    public void remove(EventType eventType) {
        mappedEvents.remove(eventType.getId());
    }
}

File: memento/src/main/java/com/alexstyl/specialdates/addevent/SelectedEvents.kt
package com.alexstyl.specialdates.addevent


import com.alexstyl.specialdates.date.Date
import com.alexstyl.specialdates.events.Event
import com.alexstyl.specialdates.events.peopleevents.EventType

class SelectedEvents
internal constructor() {

    private val mappedEvents: HashMap<Int, Event>

    val events: Collection<Event>
        get() = mappedEvents.values

    init {
        mappedEvents = HashMap()
    }

    internal fun replaceDate(eventType: EventType, date: Date) {
        val eventId = eventType.id
        val event = Event(eventType, date)
        mappedEvents[eventId] = event
    }

    fun remove(eventType: EventType) {
        mappedEvents.remove(eventType.id)
    }
}
